File: tools/commonist/src/scala/commonist/api/UploadCallback.scala
package commonist.api

trait UploadCallback {
	def progress(bytes:Long)
	
	/** when true, the file overwrites another file */
	def ignoreFileexists():Boolean
	
	/** if true, previously uploaded and deleted files will be uploaded nevertheless */
	def ignoreFilewasdeleted():Boolean
	
	/** if true, already existing files will be uploaded nevertheless */
	def ignoreDuplicate():Boolean
	
	/** if true, already existing files will be uploaded nevertheless */
	def ignoreDuplicateArchive():Boolean
}

File: tools/commonist-java/src/net/psammead/mwapi/ui/UploadCallback.java
package net.psammead.mwapi.ui;

/** callback for the FileUpload action */
public interface UploadCallback {
	/** when true, the file overwites another file */
	boolean ignoreFileexists();
	
	/** when non-null the result is used as new file name */
	String renameFileexists();
	
	/** if true, a large file will be written */
	boolean ignoreLargefile();

	/** if true, previously uploaded and deleted files will be uploaded nevertheless */
	boolean ignoreFilewasdeleted();
}
